helper/forwarding/util.go

```diff
@@ -178,6 +178,7 @@ func NewRPCResponseWriter() *RPCResponseWriter {
 		statusCode: 200,
 	}
 	//w.header.Set("Content-Type", "application/octet-stream")
+	w.Header().Set("Cache-Control", "no-store")
```

Comment:
You are right. I had overlooked the part where the headers were copied after serving the request. The embedded handler will have the headers set. Removed this line of code.